views/dashboard/about-page/components/FormView.tsx

```diff
@@ -0,0 +1,201 @@
+import { use, useEffect } from 'react'
+import { useDispatch } from 'react-redux'
+import useForm from '@/utils/hooks/useForm'
+import FormValidations from '@/utils/validations/forms'
+import { FileUpload, FormButtonGroup, TextField } from '@/components/form'
+import { updateAboutPageData } from '@/lib/db/cms/about-page'
+import { setAbout } from '@/lib/redux/slices/aboutPage'
+import { StyledForm } from './StyledFormView'
+import { handleCloudUpload } from '@/utils/fileConversions'
+import { enqueueSnackbar } from 'notistack'
+import { statuses, messages } from '@/utils/banners'
+
+const initialState = {
+    name: { value: '', error: '' },
+    description: { value: '', error: '' },
+    header: { value: '', error: '' },
+    subHeader: {
+        value: '',
+        error: ''
+    },
+    img: [
+        {
+            name: '',
+            src: '',
+            title: '',
+            type: ''
+        }
+    ],
+    role: {
+        value: '',
+        error: ''
+    },
+    title: {
+        value: '',
+        error: ''
+    }
+}
+
+const validations = {
+    name: FormValidations.yolo,
+    description: FormValidations.yolo,
+    header: FormValidations.yolo,
+    subHeader: FormValidations.yolo,
+    img: FormValidations.yolo,
+    role: FormValidations.yolo,
+    title: FormValidations.yolo
+}
+
+const FormView = ({ FormData }: any) => {
+    const dispatch = useDispatch()
+
+    const { description, header, subHeader, img, role, title, name, id } =
+        FormData
+
+    const {
+        disableSubmit,
+        form,
+        formLoading,
+        handleChange,
+        handleFormSubmit,
+        handleNonFormEventChange,
+        handleImport,
+        isDataImported
+    } = useForm(initialState, validations, FormData)
+
+    const onFormSubmit = async () => {
+        try {
+            const cloudImg = await handleCloudUpload({
+                base64: form.img.value[0].base64,
+                context: 'aboutPage',
+                filename: form.img.value[0].name
+            })
+
+            const payload = {
+                ...(description !== form.description.value && {
+                    description: form.description.value
+                }),
+                ...(header !== form.header.value && {
+                    header: form.header.value
+                }),
+                ...(subHeader !== form.subHeader.value && {
+                    subHeader: form.subHeader.value
+                }),
+                ...(role !== form.role.value && { role: form.role.value }),
+                ...(title !== form.title.value && { title: form.title.value }),
+                ...(name !== form.name.value && { name: form.name.value }),
+                ...(img !== form.img.value && {
+                    src: cloudImg,
+                    type: form.img.value[0].type,
+                    name: form.img.value[0].name
```

Comment:
While this is all good and stuff, there is definitely a better way to do this.

```
const payload = {}
for (const [key, value] of Object.entries(FormData)) {
 if (FormData[key] !== form[key].value) {
  if (key === 'img') {
     payload[key] = {
                          src: await handleCloudUpload({
                base64: form.img.value[0].base64,
                context: 'aboutPage',
                filename: form.img.value[0].name
            }),
                    type: form.img.value[0].type,
                    name: form.img.value[0].name
     }
   } else {
   payload[key] = form[key].value
}
 }
}

Reply:
Idk if this is the place to talk about it but they gotta take the garage opener away from that kid.

Reply:
id like to take this time to apologize . youre right that function you had done worked well . i couldnt see anything wrong with it unless i tried too . thats not a good trait to have . i realized that you need to just appreciate what people do when they give it their best even if they can be better . i mean look i finally use a period right.